### PR TITLE
Generate config.autoload_lib(...) for new apps

### PR DESCRIPTION
**Pull request by fxn**
With this patch, newly generated `config/application.rb` now include

```ruby
config.autoload_lib(ignore: %w(assets tasks))
```

In practice, this means that new 7.1 applications autoload from `lib` out of the box.

I considered computing the list of ignored subdirectories dynamically, or abstracting them somehow. But the simplicity and clarity of the code in this patch is hard to beat for my taste.

### CODE REVIEW

**Comment by adrianthedev** [comment]
Just to eliminate all confusion, this is the `/lib` directory from the root of your application, not `/app/lib`.

**Comment by fxn** [comment]
@adrianthedev confirm! That is what `config.autoload_lib` does (`app/lib` does not even exist).

**Comment by adrianthedev** [comment]
Yes. I know there was a bit of talk about adding an `app/lib` directory to rails and add that to the autoload paths.

**Comment by fxn** [comment]
@adrianthedev no, that won't happen.

Let me elaborate :).

If an application had `app/lib`, the directory was added automatically to the autoload paths already, since all subdirectories of `app` get added automatically except for the ones for assets, views, or JavaScript. That is why that non-standard directory started to show up. But what we have done is to fix the root cause of why `app/lib` became a thing in the first place, which is to restore autoloading from `lib`. This is now reasonable with Zeitwerk (it was not with the previous autoloader).

There are some personal remarks about `app/lib` vs `lib` in the PR that introduced `config.autoload_lib` (https://github.com/rails/rails/pull/48572).

**Comment by adrianthedev** [comment]
Yes. I think that was the discussion I was remembering.
Thank you!

**Comment by skipkayhil** [comment]
@tekin, see the discussion in palkan/view_component-contrib#39

**Comment by tekin** [comment]
Makes sense, thanks.

> @tekin, see the discussion in [palkan/view_component-contrib#39](https://github.com/palkan/view_component-contrib/issues/39)

File: railties/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+*   `config/application.rb` now includes
+
+    ```ruby
+    config.autoload_lib(ignore: %w(assets tasks))
+    ```
+
+    In practice, this means that new 7.1 applications autoload from `lib` out of the box.
```
Comment (exterm) [inline railties/CHANGELOG.md:7]: Very late nitpick:

> autoload from `lib`

This wording in the docs is an indication that `autoload_from_lib` would be a better name for the method.

Comment (fxn) [reply railties/CHANGELOG.md:7]: That wording is not so unique. We also say "autoload the `lib` directory", or "the `app/models` directory is autoloaded by default".

**Pull request merged**